gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -566,9 +633,7 @@ gst_tensor_trainer_chain (GstPad * sinkpad, GstObject * parent,
       Scheduling with subplugin does not work.
    */
   if (trainer->total_push_data_cnt == 1
-      || trainer->total_push_data_cnt ==
-      trainer->prop.num_training_samples +
-      trainer->prop.num_validation_samples) {
+      || gst_tensor_trainer_epochs_is_complete (trainer)) {
```

Comment:
In the case of ```epoch_is_complete```, I think it would be good to keep the name because it is used as below in the code. However, the code was modified as follows.
```
static gboolean
gst_tensor_trainer_epochs_is_complete (GstTensorTrainer * trainer)
{
  int required_sample;

  g_return_val_if_fail (trainer != NULL, FALSE);
  g_return_val_if_fail (trainer->fw != NULL, FALSE);
  g_return_val_if_fail (&trainer->prop != NULL, FALSE);

  required_sample =
      trainer->prop.num_training_samples + trainer->prop.num_validation_samples;
  if (trainer->total_push_data_cnt % required_sample != 0)
    return FALSE;

  gst_tensor_trainer_wait_for_epoch_completion (trainer);

  return TRUE;
}
```